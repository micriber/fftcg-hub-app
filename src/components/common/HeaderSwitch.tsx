import React from 'react';
import {Keyboard, StyleSheet, View} from 'react-native';
import {Switch, useTheme} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {ColorTranslator} from 'colortranslator';
import themes from './../../theme';

type Props = {
  leftIconName?: string;
  rightIconName?: string;
  value: boolean;
  onValueChange: (value: boolean) => void;
};

const HeaderSwitch = ({
  leftIconName,
  rightIconName,
  value,
  onValueChange,
}: Props) => {
  const theme = useTheme(themes.primary);
  return (
    <View style={styles.header}>
      {leftIconName && (
        <Icon name={leftIconName} color={theme.colors.lightGrey} size={20} />
      )}
      <Switch
        value={value}
        onValueChange={(withValue: boolean) => {
          Keyboard.dismiss();
          onValueChange(withValue);
        }}
        style={styles.switch}
        testID={`SwitcherState-${+value}`}
        thumbColor={value ? theme.colors.active : '#ffffff'}
        trackColor={{
          false: new ColorTranslator(theme.colors.darkGrey).setA(0.3).RGBA,
          true: new ColorTranslator(theme.colors.active).setA(0.5).RGBA,
        }}
      />
      {rightIconName && (
        <Icon name={rightIconName} color={theme.colors.lightGrey} size={22} />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
    marginLeft: 10,
    marginRight: 5,
  },
  switch: {
    marginLeft: 2,
  },
});

export default HeaderSwitch;
