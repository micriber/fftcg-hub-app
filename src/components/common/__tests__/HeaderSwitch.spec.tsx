/**
 * @format
 */

import 'react-native';
import React from 'react';
import HeaderSwitch from '../HeaderSwitch';
import {render} from '@testing-library/react-native';

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] HeaderSwitch', () => {
  it('renders correctly with switch false', async () => {
    const root = render(
      <HeaderSwitch
        leftIconName="view-grid"
        rightIconName="format-list-bulleted"
      />,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });
});
