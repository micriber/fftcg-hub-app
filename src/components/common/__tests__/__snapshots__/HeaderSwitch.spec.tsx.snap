// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] HeaderSwitch Switch value change 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 5,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰕰
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="rgba(61,99,145,0.5)"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "marginLeft": 2,
        },
      ]
    }
    testID="SwitcherState-0"
    thumbTintColor="#ffffff"
    tintColor="rgba(148,174,206,0.3)"
    value={false}
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 22,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰉹
  </Text>
</View>
`;

exports[`[Component] HeaderSwitch Switch value change 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 5,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰕰
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="rgba(61,99,145,0.5)"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "marginLeft": 2,
        },
      ]
    }
    testID="SwitcherState-1"
    thumbTintColor="#3D6391"
    tintColor="rgba(148,174,206,0.3)"
    value={true}
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 22,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰉹
  </Text>
</View>
`;

exports[`[Component] HeaderSwitch renders correctly with switch false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 5,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰕰
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="rgba(61,99,145,0.5)"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "marginLeft": 2,
        },
      ]
    }
    testID="SwitcherState-0"
    thumbTintColor="#ffffff"
    tintColor="rgba(148,174,206,0.3)"
    value={false}
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 22,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰉹
  </Text>
</View>
`;

exports[`[Component] HeaderSwitch renders correctly with switch right 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 5,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰕰
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="rgba(61,99,145,0.5)"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "marginLeft": 2,
        },
      ]
    }
    testID="SwitcherState-1"
    thumbTintColor="#3D6391"
    tintColor="rgba(148,174,206,0.3)"
    value={true}
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#D1DEEE",
          "fontSize": 22,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰉹
  </Text>
</View>
`;
